src/Components/AllProducts.js

```diff
@@ -1,17 +1,25 @@
 import React from 'react'
 import Individual_product from "./Individual-product";
 import './comp.css';
-
+import useFetchUserData from '../Hooks/useFetchUserData';
 function AllProducts() {
+  const {isLoading,data}=useFetchUserData('product-details2','https://obscure-refuge-62167.herokuapp.com/products');
```

Comment:
the args passed to this hook are internal to the hook and shouldn't be passed as args. You just have to expect your hook to handle all the calls, URLs...